File: OOMTest-Xamarin/OOMTest/ImageFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.Content;
using Android.Graphics;
using Android.Graphics.Drawables;
using Android.OS;
using Android.Support.V4.App;
using Android.Views;
using Android.Widget;

namespace OOMTest
{
	public class ImageFragment : Fragment
	{
		private ImageView mImageView;

		public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			var layout = (RelativeLayout) inflater.Inflate(Resource.Layout.fragment_image, container, false);

			mImageView = layout.FindViewById<ImageView>(Resource.Id.image_view);
			//using(var bitmap = Bitmap.CreateBitmap(2000, 2000, Bitmap.Config.Argb8888)) {
			//	bitmap.EraseColor(Color.Red);
			//	mImageView.SetImageBitmap(bitmap);
			//}

			var bitmap = Bitmap.CreateBitmap(2000, 2000, Bitmap.Config.Argb8888);
			bitmap.EraseColor(Color.Red);
			mImageView.SetImageBitmap(bitmap);

			return layout;
		}

		public override void OnStop()
		{
			base.OnStop();
			//((BitmapDrawable) mImageView.Drawable).Bitmap.Recycle();
			//((BitmapDrawable) mImageView.Drawable).Bitmap.Dispose();
			//mImageView.SetImageBitmap(null);
			//mImageView.Dispose();
			GC.Collect();
		}
	}
}

File: OOMTest-Android/app/src/main/java/com/tobishiba/oomtest/ImageFragment.java
package com.tobishiba.oomtest;

import android.graphics.Bitmap;
import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;

/**
 * Created by tobiasbuchholz on 12/09/16
 */
public class ImageFragment extends Fragment {
    private ImageView mImageView;

    @Nullable
    @Override
    public View onCreateView(final LayoutInflater inflater, @Nullable final ViewGroup container, @Nullable final Bundle savedInstanceState) {
        final RelativeLayout layout = (RelativeLayout) inflater.inflate(R.layout.fragment_image, container, false);

        mImageView = (ImageView) layout.findViewById(R.id.image_view);
        final Bitmap bitmap = Bitmap.createBitmap(2000, 2000, Bitmap.Config.ARGB_8888);
        bitmap.eraseColor(Color.RED);
        mImageView.setImageBitmap(bitmap);

        return layout;
    }
}
